src/community/features-templating/features-templating-core/src/test/java/org/geoserver/featurestemplating/builders/impl/DynamicIncludeFlatBuilderTest.java

```diff
@@ -0,0 +1,126 @@
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import net.sf.json.JSONObject;
+import net.sf.json.JSONSerializer;
+import org.geoserver.featurestemplating.configuration.TemplateIdentifier;
+import org.geoserver.featurestemplating.readers.RecursiveJSONParser;
+import org.geoserver.featurestemplating.writers.GeoJSONWriter;
+import org.geoserver.platform.resource.FileSystemResourceStore;
+import org.geotools.data.DataTestCase;
+import org.geotools.feature.simple.SimpleFeatureBuilder;
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.geotools.jdbc.JDBCDataStore;
+import org.junit.Before;
+import org.junit.Test;
+import org.opengis.feature.Feature;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.xml.sax.helpers.NamespaceSupport;
+
+public class DynamicIncludeFlatBuilderTest extends DataTestCase {
+
+    private static final String BASE =
+            "{\"metadata\":{\"metadata_iso_19139\":{\"title\":\"basetext\",\"href\":\"basehref\",\"type\":\"basetype\"}}}";
+    private static final String DYNAMIC_METADATA =
+            "{\"metadata\":{\"metadata_iso_19139\":{\"title\":\"metadata_iso_19139\",\"dynamicValue\":\"${dynamicValue}\",\"href\":\"http://metadata_iso_19139.org\",\"type\":\"metadata\"}}}";
+    private static final String STATIC_METADATA =
+            "{\"metadata\":{\"metadata_iso_19139\":{\"title\":\"metadata_iso_19139\",\"href\":\"http://metadata_iso_19139.org\",\"type\":\"metadata\"}}}";
+
+    private JsonNode node;
+    private SimpleFeature jsonFieldSimpleFeature;
+    FileSystemResourceStore store;
+
+    @Before
+    public void setup() throws Exception {
+        ObjectMapper mapper = new ObjectMapper();
+        node = mapper.readTree(BASE);
+
+        SimpleFeatureTypeBuilder tb = new SimpleFeatureTypeBuilder();
+        tb.userData(JDBCDataStore.JDBC_NATIVE_TYPENAME, "json");
+        tb.add("dynamicMetadata", String.class);
+        tb.add("staticMetadata", String.class);
+        tb.add("dynamicValue", String.class);
+        tb.add("arrayMetadata", String.class);
+        tb.setName("jsonFieldSimpleType");
+        SimpleFeatureType schema = tb.buildFeatureType();
+        schema.getDescriptor("dynamicMetadata")
+                .getUserData()
+                .put(JDBCDataStore.JDBC_NATIVE_TYPENAME, "json");
+        schema.getDescriptor("staticMetadata")
+                .getUserData()
+                .put(JDBCDataStore.JDBC_NATIVE_TYPENAME, "json");
+
+        SimpleFeatureBuilder fb = new SimpleFeatureBuilder(schema);
+        fb.add(DYNAMIC_METADATA);
+        fb.add(STATIC_METADATA);
+        fb.add("dynamic value result");
+        jsonFieldSimpleFeature = fb.buildFeature("jsonFieldSimpleType.1");
+
+        store = new FileSystemResourceStore(new File("src/test/resources/jsonIncludes"));
+    }
+
+    @Test
+    public void testObtainDynamicIncludeFlatKeyword() throws IOException {
```

Comment:
I would move this test into the JsonIncludesTest or the JsonTemplateReaderTest class.

Reply:
moved into JsonIncludesTest